src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -1113,14 +1108,12 @@ bool FlexPA::prepPoint_pin_checkPoint_via_helper(
     frInstTerm* inst_term,
     const std::vector<gtl::polygon_90_data<frCoord>>& layer_polys)
 {
-  return prepPoint_pin_checkPoint_viaDir_helper(
-             ap, via, pin, inst_term, layer_polys, frDirEnum::E)
-         || prepPoint_pin_checkPoint_viaDir_helper(
-             ap, via, pin, inst_term, layer_polys, frDirEnum::W)
-         || prepPoint_pin_checkPoint_viaDir_helper(
-             ap, via, pin, inst_term, layer_polys, frDirEnum::S)
-         || prepPoint_pin_checkPoint_viaDir_helper(
-             ap, via, pin, inst_term, layer_polys, frDirEnum::N);
+  bool some_dir_is_legal = false;
+  for (const frDirEnum dir : frDirEnumPlanar) {
+    some_dir_is_legal |= prepPoint_pin_checkPoint_viaDir_helper(
+        ap, via, pin, inst_term, layer_polys, dir);
+  }
```

Comment:
This is not equivalent in terms of efficiency as the || operator short-circuits and the loop doesn't.